table/src/main/java/org/enso/table/error/UnexpectedTypeException.java

```diff
@@ -0,0 +1,21 @@
+package org.enso.table.error;
+
+/** An error thrown when a type error is encountered. */
+public class UnexpectedTypeException extends RuntimeException {
+  private final String expected;
+
+  /**
+   * Creates a new instance of this error.
+   *
+   * @param expected the expected type description
+   */
+  public UnexpectedTypeException(String expected) {
```

Comment:
Is there a path at all to replacing this with the language's `Type_Error` atom?

Reply:
Not really, if we want to go in that direction, we'll have to catch these in Enso and rethrow as Type_Error